Scripts/RemediationScripts/Remediate-EnableSSLDBForMySQLFlexibleServer.ps1

```diff
@@ -0,0 +1,804 @@
+
+        .DESCRIPTION
+        Rolls back remediation done for 'Azure_DBForMySQLFlexibleServer_DP_Enable_SSL_Trial' Control.
+        Change SSL to Previous Value on Azure Database for MySQL flexible server(s) in the Subscription. 
+        
+        .PARAMETER SubscriptionId
+        Specifies the ID of the Subscription that was previously remediated.
+        
+        .PARAMETER Force
+        Specifies a forceful roll back without any prompts.
+        
+        .Parameter PerformPreReqCheck
+        Specifies validation of prerequisites for the command.
+      
+        .PARAMETER FilePath
+        Specifies the path to the file to be used as input for the roll back.
+
+        .INPUTS
+        None. You cannot pipe objects to Set-RequireSecureTransporttoPreviousValueforDBForMySQLFlexibleServer.
+
+        .OUTPUTS
+        None. Set-RequireSecureTransporttoPreviousValueforDBForMySQLFlexibleServer does not return anything that can be piped and used as an input to another command.
+
+        .EXAMPLE
+        PS> Set-RequireSecureTransporttoPreviousValueforDBForMySQLFlexibleServer -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -FilePath C:\AzTS\Subscriptions\00000000-xxxx-0000-xxxx-000000000000\202109131040\SetSSLForDBForMySQLFlexibleServer\RemediatedDBForMySQLFS.csv
+
+        .LINK
+        None
+    #>
+
+    param (
+        [String]
+        [Parameter(Mandatory = $true, HelpMessage = "Specifies the ID of the Subscription that was previously remediated.")]
+        $SubscriptionId,
+
+        [Switch]
+        [Parameter(HelpMessage = "Specifies a forceful roll back without any prompts")]
+        $Force,
+
+        [Switch]
+        [Parameter(HelpMessage = "Specifies validation of prerequisites for the command")]
+        $PerformPreReqCheck,
+
+        [String]
+        [Parameter(Mandatory = $true, HelpMessage = "Specifies the path to the file to be used as input for the roll back")]
+        $FilePath
+    )
+
+    if ($PerformPreReqCheck) {
+        try {
+            Write-Host "[Step 1 of 3] Validate and install the modules required to run the script and validate the user..."
+            Write-Host $([Constants]::SingleDashLine)
+            Write-Host "Setting up prerequisites..." -ForegroundColor $([Constants]::MessageType.Info)	
+            Write-Host $([Constants]::SingleDashLine)
+            Setup-Prerequisites
+            Write-Host "Completed setting up prerequisites"	
+            Write-Host $([Constants]::SingleDashLine)
+        }
+        catch {
+            Write-Host "Error occurred while setting up prerequisites. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host $([Constants]::DoubleDashLine)	
+            return
+        }
+    }
+    else {
+        Write-Host "[Step 1 of 3] Validate the user..." 
+        Write-Host $([Constants]::SingleDashLine)
+    }  
+
+    # Connect to Azure account
+    $context = Get-AzContext
```

Comment:
# Connect to Azure account
    $context = Get-AzContext

    if ([String]::IsNullOrWhiteSpace($context)) {
        Write-Host "Connecting to Azure account..." -ForegroundColor $([Constants]::MessageType.Info)
        Write-Host $([Constants]::SingleDashLine)
        Connect-AzAccount -Subscription $SubscriptionId -ErrorAction Stop | Out-Null
        Write-Host "Connected to Azure account." -ForegroundColor $([Constants]::MessageType.Update)        
        Write-Host $([Constants]::SingleDashLine)
    }
    # Setting up context for the current Subscription.
    $context = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop  


this block as used in remediation function can be applied here as well, right?

Reply:
The result will be same, just to make things a little consistent.